tests/run/test_Saver.py

```diff
@@ -0,0 +1,113 @@
+import copy
+from tempfile import TemporaryDirectory
+
+import pytest
+import torch
+from hypothesis import given
+from hypothesis import settings
+from hypothesis import strategies as st
+from myrtlespeech.model.seq_to_seq import SeqToSeq
+from myrtlespeech.protos import task_config_pb2
+from myrtlespeech.run.callbacks.callback import CallbackHandler
+from myrtlespeech.run.run import Saver
+from myrtlespeech.run.train import fit
+
+from tests.builders.test_task_config import build_and_check_task_config
+from tests.protos.test_task_config import task_configs
+from tests.utils.utils import check_state_dicts_match
+
+# Utilities -------------------------------------------------------------------
+
+
+def change_seq_to_seq_state_dict(seq_to_seq: SeqToSeq):
+    """Alters SeqToSeq state_dict.
+
+    This should ensure that calling this function multiple times should result
+    in a unique state_dict for **every** successive call.
+    """
+    with torch.no_grad():
+        for name, param in seq_to_seq.model.named_parameters():
+            param += 1.0
+    if seq_to_seq.optim:
+        for param_group in seq_to_seq.optim.param_groups:
+            param_group["lr"] += 0.2
+    if seq_to_seq.lr_scheduler:
+        seq_to_seq.lr_scheduler.base_lrs = [
+            x + 0.1 for x in seq_to_seq.lr_scheduler.base_lrs
+        ]
+        seq_to_seq.lr_scheduler._step_count += 2
+
+
+# Tests -----------------------------------------------------------------------
+
+
+@settings(deadline=3000)
+@given(seq_to_seq_cfg=task_configs(), check_with_fit_api=st.booleans())
+def test_seq_to_seq_correctly_built_saved_and_loaded(
+    seq_to_seq_cfg: task_config_pb2.TaskConfig, check_with_fit_api: bool,
+) -> None:
+    """Test that SeqToSeq is correctly built + saved + loaded."""
+    seq_to_seq, epochs, train_loader, _ = build_and_check_task_config(
+        seq_to_seq_cfg
+    )
+
+    with TemporaryDirectory() as tmpdir:
```

Comment:
pytest has a built-in fixture for temporary directories: https://docs.pytest.org/en/latest/tmpdir.html

Reply:
I was using this originally but `tmpdir` interacts in with hypothesis in an undesirable way. Namely,  a single `tmpdir` is shared between all hypothesis calls to the same function. In this case that causes the test to fail when it shouldn't as the directory contents affect `Saver`s loading behaviour.  

I could add teardown code but I think that removes the simplicity that `tmpdir` usually provides.